### PR TITLE
Feature/darus integration

### PR DESCRIPTION
**Pull request by zimmerml**
Enables the publishing of ROARs to the DaRus Data Repository.

- Contributor: Michael Zimmermann, @zimmerml

- [x] Ensure that you followed our [toolchain guide](https://github.com/eclipse/winery/blob/main/docs/dev/github-workflow.md#github---prepare-final-pull-request).
- [x] Branch name checked. That means, the branch name starts with `thesis/`, `fix/`, ...
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Ensure to use auto format in **all** files
- [ ] Ensure that you appear in `NOTICE` at Copyright Holders
- [ ] Tests created for changes
- [ ] Documentation updated (if needed)
- [ ] Screenshots added (for UI changes)

### CODE REVIEW

File: docker/winery.yml.tpl [hunk 1]
```diff
@@ -24,6 +24,10 @@ ui:
     eclipseChe: {{ .Env.CHE_URL_PROTOCOL }}://{{ .Env.CHE_URL }}
     tops: http://{{ .Env.TOPS_HOSTNAME }}:{{ .Env.TOPS_PORT }}
     deploymentNormalizationAssistant: http://{{ .Env.DNA_HOSTNAME }}:{{ .Env.DNA_PORT }}/{{ .Env.DNA_CONTEXT }}
+  darus:
+    server: http://{{ .Env.DARUS_HOSTNAME }}
```
Comment (lharzenetter) [inline docker/winery.yml.tpl:28]: Are you sure, that it is http?

File: org.eclipse.winery.frontends/app/tosca-management/src/app/instance/instanceHeader/researchObjectArchiveUploader.service.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                private iService: InstanceService) {
+        this.roService.setBaseUrl(this.iService.path);
+    }
+
+    uploadROAR(toscaComponent: ToscaComponent, privacyOption: string): Observable<string> {
+
+        const path = this.iService.path + '/publishROAR';
+        const headers = new HttpHeaders().set('Content-Type', 'application/json');
+
+        return this.http.post<HttpResponse<any>>(path,
```
Comment (lharzenetter) [inline org.eclipse.winery.frontends/app/tosca-management/src/app/instance/instanceHeader/researchObjectArchiveUploader.service.ts:41]: Don't you know the type of the return value?

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/researchObject/ResearchObjectArchiveUploader.java [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+                case "publication":
+                    ResearchObject.Publication pub = ro.getPublication();
+                    if (pub != null) {
+                        Iterator<JsonNode> publicationFields = field.at("/value/0").elements();
+                        while (publicationFields.hasNext()) {
+                            JsonNode publicationField = publicationFields.next();
+                            String value;
+                            switch (publicationField.get("typeName").textValue()) {
+                                case "publicationCitation":
+                                    value = pub.getCitation();
```
Comment (lharzenetter) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/researchObject/ResearchObjectArchiveUploader.java:208]: Please add curly bracer here and for the ifs

**Review by lharzenetter — COMMENTED** [review]
LGTM besides minor issues

File: docker/winery.yml.tpl (hunk 1 above)
Comment (zimmerml) [reply docker/winery.yml.tpl]: Good catch!

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/researchObject/ResearchObjectArchiveUploader.java (hunk 3 above)
Comment (zimmerml) [reply org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/researchObject/ResearchObjectArchiveUploader.java]: Simplified the whole section

**Comment by zimmerml** [comment]
Follow up at https://github.com/eclipse/winery/pull/682

**Pull request closed without merging**